feat(wallet)_: add new API to restart the wallet reload timer

belalshehab opened the pull request:
- Added `restartWalletReloadTimer` method in `api.go` to expose an API for restarting the wallet reload timer.
- Implemented `Restart` method in `reader.go` to stop and start the wallet reader.
- Updated reader_test.go to comply with minimum test coverage threshold

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (55)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | 095c9198db33fe5868b9a47e8b3d3b54b7ad4cee | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5461/1/) | 2024-07-02 01:00:54 | ~1 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5461/1/consoleText) |
| :x: | 095c9198db33fe5868b9a47e8b3d3b54b7ad4cee | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5461/1/) | 2024-07-02 01:01:15 | ~1 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5461/1/consoleText) |
| :x: | 095c9198db33fe5868b9a47e8b3d3b54b7ad4cee | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5461/1/) | 2024-07-02 01:01:26 | ~1 min | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5461/1/consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | 095c9198 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/1/) | 2024-07-02 01:01:06 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/1//consoleText) |
| | | | | | | |
| :x: | e2c703fa9caf342172093a734a071b0acd3c4ba8 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5461/2/) | 2024-07-02 01:02:42 | ~26 sec | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5461/2/consoleText) |
| :x: | e2c703fa9caf342172093a734a071b0acd3c4ba8 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5461/2/) | 2024-07-02 01:02:46 | ~31 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5461/2/consoleText) |
| :x: | e2c703fa9caf342172093a734a071b0acd3c4ba8 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5461/2/) | 2024-07-02 01:03:01 | ~45 sec | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5461/2/consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | e2c703fa | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/2/) | 2024-07-02 01:04:10 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/2//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 832b0c56221f78cc740a64bf0de3254892bc65ac | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5461/3/) | 2024-07-02 17:15:17 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240702-171203-832b0c56221f78cc740a64bf0de3254892bc65ac-pr5461.zip) |
| :heavy_check_mark: | 832b0c56221f78cc740a64bf0de3254892bc65ac | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5461/3/) | 2024-07-02 17:16:11 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240702-171203-832b0c56221f78cc740a64bf0de3254892bc65ac-pr5461.zip) |
| :heavy_check_mark: | 832b0c56221f78cc740a64bf0de3254892bc65ac | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5461/3/) | 2024-07-02 17:16:44 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240702-171203-832b0c56221f78cc740a64bf0de3254892bc65ac-pr5461.aar) |
| | | | | | | |
| :heavy_multiplication_x: | 832b0c56 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/3/) | 2024-07-02 17:15:37 | ~3 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/3//consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | 58ccc28c | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/4/) | 2024-07-02 17:23:20 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/4//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 58ccc28cf87ea7ea9526e7fae60a933e815a2fa4 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5461/4/) | 2024-07-02 17:23:49 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240702-172208-58ccc28cf87ea7ea9526e7fae60a933e815a2fa4-pr5461.aar) |
| :heavy_check_mark: | 58ccc28cf87ea7ea9526e7fae60a933e815a2fa4 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5461/4/) | 2024-07-02 17:24:15 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240702-172212-58ccc28cf87ea7ea9526e7fae60a933e815a2fa4-pr5461.zip) |
| :heavy_check_mark: | 58ccc28cf87ea7ea9526e7fae60a933e815a2fa4 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5461/4/) | 2024-07-02 17:24:46 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240702-172212-58ccc28cf87ea7ea9526e7fae60a933e815a2fa4-pr5461.zip) |
| | | | | | | |
| :heavy_check_mark: | b12a3f41fff7ffb1cbea314cc10f99e9b3662cb1 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5461/5/) | 2024-07-02 17:25:51 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240702-172405-b12a3f41fff7ffb1cbea314cc10f99e9b3662cb1-pr5461.aar) |
| :heavy_check_mark: | b12a3f41fff7ffb1cbea314cc10f99e9b3662cb1 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5461/5/) | 2024-07-02 17:26:36 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240702-172432-b12a3f41fff7ffb1cbea314cc10f99e9b3662cb1-pr5461.zip) |
| :heavy_check_mark: | b12a3f41fff7ffb1cbea314cc10f99e9b3662cb1 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5461/5/) | 2024-07-02 17:28:39 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240702-172524-b12a3f41fff7ffb1cbea314cc10f99e9b3662cb1-pr5461.zip) |
| | | | | | | |
| :heavy_multiplication_x: | b12a3f41 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/5/) | 2024-07-02 17:25:53 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/5//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 2b51ead297179884a817c16d79635acbced44688 | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5461/6/) | 2024-07-03 00:13:10 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240703-001045-2b51ead297179884a817c16d79635acbced44688-pr5461.aar) |
| :heavy_check_mark: | 2b51ead297179884a817c16d79635acbced44688 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5461/6/) | 2024-07-03 00:13:14 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240703-001049-2b51ead297179884a817c16d79635acbced44688-pr5461.zip) |
| :heavy_check_mark: | 2b51ead297179884a817c16d79635acbced44688 | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5461/6/) | 2024-07-03 00:14:02 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240703-001045-2b51ead297179884a817c16d79635acbced44688-pr5461.zip) |
| | | | | | | |
| :heavy_multiplication_x: | 2b51ead2 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/6/) | 2024-07-03 00:13:25 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/6//consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | bd52fc68 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/7/) | 2024-07-03 00:30:22 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/7//consoleText) |
| | | | | | | |
| :heavy_check_mark: | bd52fc68416ec16e73b19d4b7ed601f9791a8975 | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5461/7/) | 2024-07-03 00:30:39 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240703-002900-bd52fc68416ec16e73b19d4b7ed601f9791a8975-pr5461.aar) |
| :heavy_check_mark: | bd52fc68416ec16e73b19d4b7ed601f9791a8975 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5461/7/) | 2024-07-03 00:31:06 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240703-002900-bd52fc68416ec16e73b19d4b7ed601f9791a8975-pr5461.zip) |
| :heavy_check_mark: | bd52fc68416ec16e73b19d4b7ed601f9791a8975 | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5461/7/) | 2024-07-03 00:32:01 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240703-002900-bd52fc68416ec16e73b19d4b7ed601f9791a8975-pr5461.zip) |
| | | | | | | |
| :heavy_multiplication_x: | 8cb1dce6 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/8/) | 2024-07-03 00:36:51 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/8//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 8cb1dce6b788df3dc265da06d1bb18af570642cd | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5461/8/) | 2024-07-03 00:37:26 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240703-003535-8cb1dce6b788df3dc265da06d1bb18af570642cd-pr5461.aar) |
| :heavy_check_mark: | 8cb1dce6b788df3dc265da06d1bb18af570642cd | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5461/8/) | 2024-07-03 00:37:47 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240703-003539-8cb1dce6b788df3dc265da06d1bb18af570642cd-pr5461.zip) |
| :heavy_check_mark: | 8cb1dce6b788df3dc265da06d1bb18af570642cd | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5461/8/) | 2024-07-03 00:38:27 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240703-003539-8cb1dce6b788df3dc265da06d1bb18af570642cd-pr5461.zip) |
| | | | | | | |
| :heavy_multiplication_x: | 1d7b91e1 | [#9](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/9/) | 2024-07-03 00:44:48 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/9//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 1d7b91e116d34c15467ba0d6d129fa09a58f40be | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5461/9/) | 2024-07-03 00:45:17 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240703-004320-1d7b91e116d34c15467ba0d6d129fa09a58f40be-pr5461.aar) |
| :heavy_check_mark: | 1d7b91e116d34c15467ba0d6d129fa09a58f40be | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5461/9/) | 2024-07-03 00:46:23 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240703-004324-1d7b91e116d34c15467ba0d6d129fa09a58f40be-pr5461.zip) |
| :heavy_check_mark: | 1d7b91e116d34c15467ba0d6d129fa09a58f40be | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5461/9/) | 2024-07-03 00:47:36 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240703-004324-1d7b91e116d34c15467ba0d6d129fa09a58f40be-pr5461.zip) |
| | | | | | | |
| :heavy_check_mark: | 528ee35edcf1e15f5d5124b1998b46b5a71438fd | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5461/10/) | 2024-07-03 09:46:05 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240703-094228-528ee35edcf1e15f5d5124b1998b46b5a71438fd-pr5461.zip) |
| :heavy_check_mark: | 528ee35edcf1e15f5d5124b1998b46b5a71438fd | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5461/10/) | 2024-07-03 09:48:12 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240703-094228-528ee35edcf1e15f5d5124b1998b46b5a71438fd-pr5461.aar) |
| :heavy_check_mark: | 528ee35edcf1e15f5d5124b1998b46b5a71438fd | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5461/10/) | 2024-07-03 09:49:29 | ~7 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240703-094228-528ee35edcf1e15f5d5124b1998b46b5a71438fd-pr5461.zip) |
| | | | | | | |
| :heavy_multiplication_x: | 528ee35e | [#10](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/10/) | 2024-07-03 09:52:44 | ~10 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/10//consoleText) |
| :heavy_check_mark: | 528ee35e | [#11](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/11/) | 2024-07-03 10:58:08 | ~42 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/11//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 58a8bc260dec2968bd370466f248f0c5c450058b | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5461/11/) | 2024-07-03 11:31:34 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240703-112853-58a8bc260dec2968bd370466f248f0c5c450058b-pr5461.zip) |
| :heavy_check_mark: | 58a8bc260dec2968bd370466f248f0c5c450058b | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5461/11/) | 2024-07-03 11:31:49 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240703-112853-58a8bc260dec2968bd370466f248f0c5c450058b-pr5461.zip) |
| :heavy_check_mark: | 58a8bc260dec2968bd370466f248f0c5c450058b | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5461/11/) | 2024-07-03 11:34:34 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240703-112849-58a8bc260dec2968bd370466f248f0c5c450058b-pr5461.aar) |
| | | | | | | |
| :heavy_multiplication_x: | 58a8bc26 | [#12](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/12/) | 2024-07-03 11:35:17 | ~6 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/12//consoleText) |
| :heavy_multiplication_x: | 58a8bc26 | [#13](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/13/) | 2024-07-03 11:48:05 | ~4 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/13//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 7aed696a57545b66b2f68014d9401c6c18596c0b | [#12](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5461/12/) | 2024-07-03 11:51:20 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240703-114935-7aed696a57545b66b2f68014d9401c6c18596c0b-pr5461.aar) |
| :heavy_check_mark: | 7aed696a57545b66b2f68014d9401c6c18596c0b | [#12](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5461/12/) | 2024-07-03 11:52:11 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240703-114939-7aed696a57545b66b2f68014d9401c6c18596c0b-pr5461.zip) |
| :heavy_check_mark: | 7aed696a57545b66b2f68014d9401c6c18596c0b | [#12](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5461/12/) | 2024-07-03 11:53:32 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240703-114939-7aed696a57545b66b2f68014d9401c6c18596c0b-pr5461.zip) |
| | | | | | | |
| :heavy_check_mark: | 7aed696a | [#14](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/14/) | 2024-07-03 12:33:51 | ~44 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/14//consoleText) |
| | | | | | | |
| :heavy_check_mark: | fa0ef90450dee2974396c2c519d032634204793f | [#13](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5461/13/) | 2024-07-03 15:46:02 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240703-154329-fa0ef90450dee2974396c2c519d032634204793f-pr5461.aar) |
| :heavy_check_mark: | fa0ef90450dee2974396c2c519d032634204793f | [#13](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5461/13/) | 2024-07-03 15:46:15 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240703-154332-fa0ef90450dee2974396c2c519d032634204793f-pr5461.zip) |
| :heavy_check_mark: | fa0ef90450dee2974396c2c519d032634204793f | [#13](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5461/13/) | 2024-07-03 15:48:15 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240703-154332-fa0ef90450dee2974396c2c519d032634204793f-pr5461.zip) |
| | | | | | | |
| :heavy_multiplication_x: | fa0ef904 | [#16](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/16/) | 2024-07-03 16:38:39 | ~13 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/16//consoleText) |
| :heavy_multiplication_x: | fa0ef904 | [#17](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/17/) | 2024-07-03 16:39:59 | ~23 sec | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/17//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | b79719fc262b58f1e8409105c60b6ae8511011f5 | [#14](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5461/14/) | 2024-07-03 16:50:27 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240703-164812-b79719fc262b58f1e8409105c60b6ae8511011f5-pr5461.zip) |
| :heavy_check_mark: | b79719fc262b58f1e8409105c60b6ae8511011f5 | [#14](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5461/14/) | 2024-07-03 16:50:34 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240703-164811-b79719fc262b58f1e8409105c60b6ae8511011f5-pr5461.zip) |
| :heavy_check_mark: | b79719fc262b58f1e8409105c60b6ae8511011f5 | [#14](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5461/14/) | 2024-07-03 16:53:36 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240703-164811-b79719fc262b58f1e8409105c60b6ae8511011f5-pr5461.aar) |
| | | | | | | |
| :heavy_check_mark: | b79719fc | [#18](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/18/) | 2024-07-03 17:31:56 | ~43 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5461/18//consoleText) |

saledjenic commented on services/wallet/reader.go:
```diff
@@ -25,9 +26,10 @@ import (
 
 const (
-	walletTickReloadPeriod      = 10 * time.Minute
+        walletTickReloadPeriod      = 3 * time.Minute
```
Why 3?

belalshehab replied:
to speed up my tests, I will switch it back to 10 when I create the final PR

belalshehab replied:
just to speed up my tests.
fixed in final PR

saledjenic commented on services/wallet/reader.go:
```diff
@@ -195,6 +198,19 @@ func (r *Reader) startWalletEventsWatcher() {
+        r.walletFeed.Subscribe(SIGNAL_MANUAL_RELOAD_TRIGGERED, handleManualReloadTriggered)
+}
+
+func handleManualReloadTriggered(event walletevent.Event) {
```
There is no receiver set for this function, but in the body, you're calling `r.`, that won't compile.

saledjenic reviewed (changes requested):
@belalshehab all you need to do here is to add a new function like this:
```
func (r *Reader) Restart() error {
	r.Stop()
	return r.Start()
}
```
Then exposed it via wallet service API, like:
```
func (api *API) RestartWalletReloadTimer(ctx context.Context) error {
	return api.s.reader.Restart()
}
```
in this file:
https://github.com/status-im/status-go/blob/develop/services/wallet/api.go

and just use it on the client side when needed.

Commit:
feat(wallet)_: add new API to restart the wallet reload timer

- Added `restartWalletReloadTimer` method in `api.go` to expose an API for restarting the wallet reload timer.
- Implemented `Restart` method in `reader.go` to stop and start the wallet reader.
- Updated reader_test.go to comply with minimum test coverage threshold

belalshehab commented:
Hi @igor-sirotin 
Do you have any suggestions on how to bypass the testCovarage requirement for this PR?
it doesn't make sense to add unit tests for this APi
